erpnext/hr/report/bank_remittance/bank_remittance.py

```diff
@@ -0,0 +1,165 @@
+			"options": "Employee",
+			"width": 200
+		},
+		{
+			"label": _("Bank Name"),
+			"fieldtype": "Data",
+			"fieldname": "bank_name",
+			"width": 50
+		},
+		{
+			"label": _("Employee A/C Number"),
+			"fieldtype": "Int",
+			"fieldname": "employee_account_no",
+			"width": 50
+		},
+		{
+			"label": _("IFSC Code"),
+			"fieldtype": "Data",
+			"fieldname": "bank_code",
+			"width": 100
+		},
+		{
+			"label": _("Currency"),
+			"fieldtype": "Data",
+			"fieldname": "currency",
+			"width": 50
+		},
+		{
+			"label": _("Net Salary Amount"),
+			"fieldtype": "Currency",
+			"options": "currency",
+			"fieldname": "amount",
+			"width": 100
+		}
+	]
+	data = []
+
+	accounts = get_account()
+	payroll_entries = get_payroll_entry(accounts, filters)
+	salary_slips = get_salary_slips(payroll_entries)
+	get_emp_bank_ifsc_code(salary_slips)
+
+	for salary in salary_slips:
+		if salary.bank_name and salary.bank_account_no and salary.debit_acc_no and salary.status in ["Submitted", "Paid"]:
+			row = {
+					"payroll_no": salary.payroll_entry,
+					"debit_account": salary.debit_acc_no,
+					"payment_date": frappe.utils.formatdate(salary.modified.strftime('%Y-%m-%d')),
+					"bank_name":salary.bank_name,
+					"employee_account_no":salary.bank_account_no,
+					"bank_code": salary.ifsc_code,
+					"employee_name": salary.employee+": " + salary.employee_name,
+					"currency": frappe.get_cached_value('Company', filters.company, 'default_currency'),
+					"amount": salary.net_pay,
+				}
+
+			data.append(row)
+	return columns, data
+
+def get_account():
+	accounts = frappe.get_all("Account",
+	filters={
+		"account_type": "Bank"
+	}, as_list =1)
+
+	return accounts
+
+def get_payroll_entry(accounts, filters):
+	accounts = list(itertools.chain(*accounts))
+	accounts = ', '.join(map(str, accounts))
+	payroll_filter = [
+			('payment_account', 'IN', accounts),
+			('number_of_employees', '>', 0),
+			('Company', '=', filters.company)
+		]
+	if filters.to_date:
+		payroll_filter.append(('posting_date', '<', filters.to_date))
+
+	if filters.from_date:
+		payroll_filter.append(('posting_date', '>', filters.from_date))
+
+	entries = get_record("Payroll Entry", payroll_filter, ["name", "payment_account"])
+
+	p = list(map(lambda x: {k:v for k, v in x.items() if k == 'payment_account'}, entries))
+	payment_accounts = ', '.join(map(str, [l['payment_account'] for l in p if 'payment_account' in l]))
+	get_company_account(payment_accounts, entries)
+	return entries
+
+def get_salary_slips(payroll_entries):
+	d = list(map(lambda x: {k:v for k, v in x.items() if k == 'name'}, payroll_entries))
+	payroll = ', '.join(map(str, [l['name'] for l in d if 'name' in l]))
+	salary_slips = get_record("Salary Slip", [("payroll_entry", "IN", payroll)],
+		fields = ["modified", "net_pay", "bank_name", "bank_account_no", "payroll_entry", "employee", "employee_name", "status"]
+	)
+
+	# appending company debit accounts
+	for slip in salary_slips:
+		for entry in payroll_entries:
```

Comment:
This is unnecessarily complicated. No need to write nested for loop.

Reply:
done